server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -106,32 +109,78 @@ public SetMessagesResponse process(SetMessagesRequest request, MailboxSession ma
         Mailbox<Id> outbox;
         try {
             outbox = getOutbox(mailboxSession).orElseThrow(() -> new MailboxRoleNotFoundException(Role.OUTBOX));
-        } catch (MailboxException | MailboxRoleNotFoundException e) {
+        } catch (MailboxRoleNotFoundException e) {
             LOGGER.error("Unable to find a mailbox with role 'outbox'!");
             throw Throwables.propagate(e);
         }
 
         // handle errors
+        Predicate<CreationMessage> notImplementedPredicate = getNotImplementedRequestTester(mailboxSession);
         Predicate<CreationMessage> validMessagesTester = CreationMessage::isValid;
         Predicate<CreationMessage> invalidMessagesTester = validMessagesTester.negate();
         SetMessagesResponse.Builder responseBuilder = SetMessagesResponse.builder()
-                .notCreated(handleCreationErrors(invalidMessagesTester, request));
+                .notCreated(handleCreationErrors(request, invalidMessagesTester, notImplementedPredicate));
 
+        // Dealing with sending messages only
         return request.getCreate().entrySet().stream()
-                .filter(e -> validMessagesTester.test(e.getValue()))
+                .filter(e -> isRequestForSending(e.getValue(), mailboxSession, validMessagesTester))
                 .map(e -> new MessageWithId.CreationMessageEntry(e.getKey(), e.getValue()))
                 .map(nuMsg -> createMessageInOutboxAndSend(nuMsg, mailboxSession, outbox, buildMessageIdFunc(mailboxSession, outbox)))
                 .map(msg -> SetMessagesResponse.builder().created(ImmutableMap.of(msg.getCreationId(), msg.getMessage())).build())
                 .reduce(responseBuilder, SetMessagesResponse.Builder::accumulator, SetMessagesResponse.Builder::combiner)
                 .build();
     }
 
-    private Map<CreationMessageId, SetError> handleCreationErrors(Predicate<CreationMessage> invalidMessagesTester,
-                                                                  SetMessagesRequest request) {
-        return request.getCreate().entrySet().stream()
-                .filter(e -> invalidMessagesTester.test(e.getValue()))
+    private boolean isRequestForSending(CreationMessage messageWithId, MailboxSession session,
+                                        Predicate<CreationMessage> validMessagesTester) {
+
+        boolean isMessageSetInOutbox = getOutbox(session)
+                .map(box -> box.getMailboxId().serialize())
+                .map(id -> messageWithId.getMailboxIds().contains(id))
+                .orElse(false);
+        return validMessagesTester.test(messageWithId) && isMessageSetInOutbox;
+    }
+
+    private Predicate<CreationMessage> getNotImplementedRequestTester(MailboxSession mailboxSession) {
+        Optional<Id> draftsId = getDraftsId(mailboxSession);
+        return e -> draftsId
```

Comment:
What is e in this case ? remember : you can give short name to things that are close enough to the code.